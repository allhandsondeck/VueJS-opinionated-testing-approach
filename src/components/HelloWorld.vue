<script setup lang="ts">
defineProps<{
  msg: string
}>()

// Pure function that can be easily moved to composable
const getTimeBasedGreeting = (hour: number): string => {
  if (hour >= 5 && hour < 12) return 'Good morning'
  if (hour >= 12 && hour < 17) return 'Good afternoon'
  if (hour >= 17 && hour < 22) return 'Good evening'
  return 'Good night'
}

// Using the pure function
const currentGreeting = getTimeBasedGreeting(new Date().getHours())
</script>

<template>
  <div class="greetings">
    <h1 class="green">{{ currentGreeting }}, {{ msg }}</h1>
    <h3>
      You've successfully created a project with
      <a href="https://vite.dev/" target="_blank" rel="noopener">Vite</a> +
      <a href="https://vuejs.org/" target="_blank" rel="noopener">Vue 3</a>.
    </h3>
  </div>
</template>
