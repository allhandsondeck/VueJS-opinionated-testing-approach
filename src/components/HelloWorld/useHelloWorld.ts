export const useHelloWorld = () => {
  const getTimeBasedGreeting = (currentHour: number): string => {
    if (currentHour >= 5 && currentHour < 12) return 'Good morning'
    if (currentHour >= 12 && currentHour < 17) return 'Good afternoon'
    if (currentHour >= 17 && currentHour < 22) return 'Good evening'
    return 'Good night'
  }

  return {
    getTimeBasedGreeting,
  }
}
