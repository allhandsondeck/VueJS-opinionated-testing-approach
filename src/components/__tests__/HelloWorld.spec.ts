import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest'
import HelloWorld from '../HelloWorld.vue'

describe('HelloWorld.vue', () => {
  it('renders message based on time of day', () => {
    vi.useFakeTimers()
    const times = [
      { time: '2023-10-10T05:00:00Z', expected: 'Good morning' },
      { time: '2023-10-10T12:00:00Z', expected: 'Good afternoon' },
      { time: '2023-10-10T18:00:00Z', expected: 'Good evening' },
      { time: '2023-10-10T23:00:00Z', expected: 'Good night' },
    ]

    times.forEach(({ time, expected }) => {
      vi.setSystemTime(new Date(time))
      const wrapper = mount(HelloWorld)
      expect(wrapper.text()).toContain(expected)
    })
    vi.useRealTimers()
  })
})
