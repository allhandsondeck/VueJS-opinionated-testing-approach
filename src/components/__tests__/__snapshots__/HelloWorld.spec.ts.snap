// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld.vue > matches snapshot 1`] = `
"<div class="greetings">
  <h1 class="green">Good evening, </h1>
  <h3> You've successfully created a project with <a href="https://vite.dev/" target="_blank" rel="noopener">Vite</a> + <a href="https://vuejs.org/" target="_blank" rel="noopener">Vue 3</a>. </h3>
</div>"
`;
